exovetter/lpp.py

```diff
@@ -0,0 +1,330 @@
+
+def lpp_onetransit(tcedata, mapInfo, ntransit):
+    """
+    Chop down the full time series to one orbital period.
+    Then gather the lpp value for that one transit.
+    """
+    
+    startTime=tcedata.time[0]+ntransit*tcedata.period
+    endTime=tcedata.time[0]+(ntransit+1)*tcedata.period + 3/24.0  #A few cadences of overlap
+    
+    want=(tcedata.time>=startTime) & (tcedata.time<=endTime)
+    newtime=tcedata.time[want]
+    newflux=tcedata.flux[want]
+    
+    nExpCad=(tcedata.time[-1]-tcedata.time[0])/tcedata.period
+    
+    if len(newtime>nExpCad*0.75):
+        onetransit=copy.deepcopy(tcedata)
+        onetransit.time=newtime
+        onetransit.flux=newflux
+        normTLpp, rawTLpp, transformedTr=computeLPPTransitMetric(onetransit,mapInfo)
+    else:
+        normTLpp=np.nan
+        rawTLpp=np.nan
+        
+    return normTLpp,rawTLpp
+
+
+def lpp_averageIndivTransit(tcedata, mapInfo):
+    """
+    
+    Create the loop over individual transits and return 
+    array normalized lpp values, mean and std.
+    Input TCE object and mapInfo object.
+    
+    It is unclear that this individual transit approach
+    separates out several new false positives.
+    It probably would require retuning for low SNR signals.
+    
+    """    
+    length=tcedata.time[-1]-tcedata.time[0]
+    ntransits=int(np.floor(length/tcedata.period))
+    
+    lppNorms=np.ones(ntransits)
+    lppRaws=np.ones(ntransits)
+    
+    #nExpCad=(tcedata.time[-1]-tcedata.time[0])/tcedata.period
+    
+    for i in range(ntransits):
+        lppNorms[i],lppRaws[i] = lpp_onetransit(tcedata,mapInfo,i)
+    
+    lppMed=np.nanmedian(lppNorms)
+    lppStd=np.nanstd(lppNorms)
+    
+    return lppNorms,lppMed, lppStd, ntransits
+
+
+#%----------------
+class Lppdata():
+    
+    def __init__(self, tce, lc, lc_name = "flux"):
+        #Expected a tce object
+        #Expecting a lightkurve object
+        #Needs a check lightcurve function
+        
+        self.check_tce(tce)
+        
+        self.tzero = tce['tzero']
+        self.dur = tce['duration']
+
+        self.time = lc.time
+        self.flux = lc.__dict__[lc_name]
+
+
+    def check_tce(self, tce):
+        
+        try:
+            self.mes = tce['snr']
+        except AttributeError:
+            print('WARNING: LPP requires a MES or SNR value stored as snr in the tce. Using a value of 10.')
+            self.mes = 10.0
+            pass
+        try:
+            self.period = tce['period']
+        except:
+            print('Exception: Period required for the TCE to run LPP.')
+            
+
+#%---------
+import scipy.io as spio
+
+class Loadmap(object):
+    
+    def __init__(self,filename):
+        
+        self.filename=filename
+        
+        self.readMatlabBlob(filename)
+
+    
+    def readMatlabBlob(self,filename):
+        """
+        read in matlab blob
+        Using the DV trained one.
+        """      
+
+        mat=spio.loadmat(filename,matlab_compatible=True)
```

Comment:
I guess this line will be replaced by what is in #18 when that one is merged, or is this something else?